_stbt/core.py

```diff
@@ -1531,12 +1545,13 @@ def push_sample(self, sample):
 
     def on_error(self, pipeline, _bus, message):
         assert message.type == Gst.MessageType.ERROR
-        Gst.debug_bin_to_dot_file_with_ts(
-            pipeline, Gst.DebugGraphDetails.ALL, "ERROR")
+        if pipeline is not None:
+            # If we receive an error after the pipeline has been torn down:
+            Gst.debug_bin_to_dot_file_with_ts(
+                pipeline, Gst.DebugGraphDetails.ALL, "ERROR")
```

Comment:
I'm a bit confused by the comment. It seems to suggest that you'll run the code inside the `if` block if we receive an error after the pipeline has been torn down.

In fact you've introduced the `if` so that `debug_bin_to_dot_file_with_ts` doesn't get run at all if we get an error after the pipeline has been torn down?

Seems clearer to just remove the comment altogether.